engine-nextgen/src/test/kotlin/graphql/nadel/validation/NadelPolymorphicHydrationValidationTest.kt

```diff
@@ -0,0 +1,229 @@
+            assert(errors.map { it.message }.isEmpty())
+        }
+
+        it("fails if return type is not a union") {
+            val fixture = NadelValidationTestFixture(
+                overallSchema = mapOf(
+                    "issues" to """
+                        type Query {
+                            issue: Issue
+                        }
+                        type Issue {
+                            id: ID!
+                            creator: User
+                            @hydrated(
+                                service: "users"
+                                field: "user"
+                                arguments: [
+                                    {name: "id", value: "$source.creatorId"}
+                                ]
+                            )
+                            @hydrated(
+                                service: "users"
+                                field: "externalUser"
+                                arguments: [
+                                    {name: "id", value: "$source.creatorId"}
+                                ]
+                            )
+                        }
+                        union AbstractUser = User | ExternalUser
+                    """.trimIndent(),
+                    "users" to """
+                        type User {
+                            id: ID!
+                            name: String!
+                        }
+
+                        type ExternalUser {
+                            id: ID!
+                            name: String!
+                        }
+                    """.trimIndent(),
+                ),
+                underlyingSchema = mapOf(
+                    "issues" to """
+                        type Query {
+                            issue: Issue
+                        }
+                        type Issue {
+                            id: ID!
+                            creatorId: ID!
+                        }
+                    """.trimIndent(),
+                    "users" to """
+                        type Query {
+                            user(id: ID!): User
+                            externalUser(id: ID!): ExternalUser
+                        }
+                        type User {
+                            id: ID!
+                            name: String!
+                        }
+                        type ExternalUser {
+                            id: ID!
+                            name: String!
+                        }
+                    """.trimIndent(),
+                ),
+            )
+
+            val errors = validate(fixture)
+            assert(errors.map { it.message }.isNotEmpty())
+            val error = errors.filterIsInstance<FieldWithPolymorphicHydrationMustReturnAUnion>().single()
+            assert(error.message == "Field Issue.creator declares a polymorphic hydration so its output type MUST be a union")
```

Comment:
I think just assert `error.parentType.name == "Issue"` and `error.overallField.name == "creator"` is better than checking the message (which can change).